versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/spi/AbstractDatabaseAdapter.java

```diff
@@ -1334,4 +1336,33 @@ protected void checkExpectedGlobalStates(
       }
     }
   }
+
+  /** Load the refLog entry for the given hash, return {@code null}, if not found. */
+  protected abstract RefLog fetchFromRefLog(OP_CONTEXT ctx, Hash refLogId);
+
+  /**
+   * Fetch multiple {@link RefLog refLog-entries} from the refLog. The returned list must have
+   * exactly as many elements as in the parameter {@code hashes}. Non-existing hashes are returned
+   * as {@code null}.
+   */
+  protected abstract List<RefLog> fetchPageFromRefLog(OP_CONTEXT ctx, List<Hash> hashes);
+
+  /** Reads from the refLog starting at the given refLog-hash. */
+  protected Stream<RefLog> readRefLogStream(OP_CONTEXT ctx, Hash initialHash)
+      throws RefLogNotFoundException {
+    Spliterator<RefLog> split = readRefLog(ctx, initialHash);
+    return StreamSupport.stream(split, false);
```

Comment:
This requires an  onClose` on the stream to delegate the close to the underlying spliterator

Reply:
I have already handled for Txn store, non-txn it is not present similar to `readCommitLogStream` because there is no releaseConnection impl.

Reply:
Ah, right.